src/main/java/net/mcreator/element/types/Painting.java

```diff
@@ -40,12 +39,12 @@ public Painting(ModElement element) {
 
 	@Override public BufferedImage generateModElementPicture() {
 		return MinecraftImageGenerator.Preview.generatePaintingPreviewPicture(getModElement().getFolderManager()
-				.getTextureFile(FilenameUtilsPatched.removeExtension(texture), TextureType.OTHER), width, height);
+				.getTextureFile(texture, TextureType.OTHER), width, height);

```

Comment:
Yep, just in case the user still uses an older workspace.

Reply:
Ok, but did you test this? `TextureHolder#setTextureFromTextureName` will be only called in the frontend (UI), so unless the mod element is actually open, no conversion will happen and this will fail to work?

Reply:
I tested using a new workspace I created in the first 2022.2 snapshot and I had no problem.

Reply:
I am struggling to understand where png extension is removed from existing workspaces when the conversion of the workspaces runs.

Reply:
![image](https://user-images.githubusercontent.com/38427877/177588641-bcc8791e-ea97-4f3e-a470-08ec19498381.png)

Reply:
Yes, but this method will not be called on workspace import. If you don't believe me, take old workspace, set breakpoint to this line and import said workspace

Reply:
> Yes, but this method will not be called on workspace import. If you don't believe me, take old workspace, set breakpoint to this line and import said workspace

I believe you and I understand what you say. However, I tested yesterday to be sure I did not make a mistake and the code seemed to tell me I was right as I did not see a problem.